lib/cylc/task_job_mgr.py

```diff
@@ -123,22 +148,73 @@ def check_task_jobs(self, suite, task_pool):
                 poll_tasks.add(itask)
         self.poll_task_jobs(suite, poll_tasks)
 
+    def init_host_exec_commands(
+            self, host, owner, user_at_host, r_suite_run_dir, cmds):
+        """Submits the commands needed to initialise the suite on
+        remote host on the first run of the suite, by sending these commands
+        to the multiprocessing pool.
+
+        Args:
+            cmds: A list of host initialisation commands (copy files etc.)
+            user_at_host: [USERNAME]@[HOSTNAME]
+        """
+        # Create init manager
+        host_init_commander = HostInitter(cmds)
+        # TODO: These might need to be executed sequentially in the pool..?
+        for cmd in cmds:
+            self.proc_pool.put_command(
+                SuiteProcContext(self.JOB_INIT_HOST, cmd),
+                self._init_host_command_callback,
+                [cmd, owner, host, user_at_host, r_suite_run_dir,
+                 host_init_commander])
```

Comment:
I think potentially there is a race condition here as the init commands could be executed out of order once in the pool? (So for instance, you wouldn't want the copy files command executing before the directory structure had been created)